static/js/components/downloads.js

```diff
@@ -22,7 +46,30 @@
 
     if (catId != catRef) {
       $(that).addClass('downloads__category--active')
+      console.log(cat);
       $(compiled({ 'catId': catId, 'catTitle': cat[0].family, 'downloads': cat })).insertAfter( container )
+
+      // Load the latest version for each device type listed in this category
+      cat.forEach(function(deviceType) {
+        getLatestVersion(deviceType.id, function(err, data) {
+          var $link = $('a.js-download-link[data-device-id="' + deviceType.id + '"]');
+          if ($link) {
+            $link.prev('i.fa').hide()
+
+            if (err) {
+              $link.parent().append('<span class="error-message">' + err + '</span>')
+            } else {
+              $link.parent().find('.error-message').remove()
+              $link.find('.os-label').text('resinOS ' + data.version)
+              $link
+                .attr('href', data.url)
+                .show()
+                .prev('i.fa').hide()
```

Comment:
If-you-are-in-the-mood: How about `.fa-cog` or `.fa-spin` to be more descriptive?